bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,936 @@
+
+    return parseFloat(parseInt(arr[0], 10) + '.' + (dec<10?'0':'') + dec);
+}
+
+async function fullSync({dataStore, client}) {
+    return getTimeEntries(dataStore, client);
+}
+
+async function getEmployeeBusinessPartner(client, user_email) {
+    let employeeDataUrl = `YY1_EMPLOYEEDATA_CDS/YY1_EmployeeData?$filter=DefaultEmailAddress eq '${user_email}'`
+
+    let employeeDataGET = await client.fetch(employeeDataUrl,
+        {
+            method: 'GET',
+            headers: headersGET
+        });
+
+    if (!employeeDataGET.ok) {
+        throw new Error(`Could not Retrieve Employee Data(${employeeDataGET.status}: ${employeeDataGET.statusText})`);
+    }
+
+    let employeeDataResponse = await employeeDataGET.json()
+
+    let employee = employeeDataResponse.d.results[0]
+
+    return employee.BusinessPartner
+}
+
+async function getWorkAgreement(client, employeeBusinessPartner) {
+    let workAgreementUrl = `YY1_WORKAGREEMENTDETAILS_CDS/YY1_WorkAgreementDetails?$filter=Person eq '${employeeBusinessPartner}'`
+
+    let workAgreementGET = await client.fetch(workAgreementUrl,
+        {
+            method: 'GET',
+            headers: headersGET
+        });
+
+    if (!workAgreementGET.ok) {
+        throw new Error(`Could not Retrieve Employee Data(${workAgreementGET.status}: ${workAgreementGET.statusText})`);
+    }
+
+    let workAgreementResponse = await workAgreementGET.json()
+
+    return workAgreementResponse.d.results[0]
+}
+
+async function incrementalSync({dataStore, client, latestSynchronizationTime}) {
+    return getTimeEntries(dataStore, client, latestSynchronizationTime);
+}
+
+async function getTimeEntries(dataStore, client, latestSynchronizationTime) {
+    let timeEntries = [];
+    let today = moment().format('YYYY-MM-DDT00:00:00')
+    let sevenDaysBefore = moment().subtract(7,'d').format('YYYY-MM-DDT00:00:00')
+    let thirtyDaysBefore = moment().subtract(30,'d').format('YYYY-MM-DDT00:00:00')
+
+    let dateFilter = `TimeSheetDate ge datetime'${thirtyDaysBefore}' and TimeSheetDate le datetime'${today}'`
+
+    if (latestSynchronizationTime) {
+        dateFilter = `TimeSheetDate ge datetime'${sevenDaysBefore}'`
+    }
+
+    let statusFilter = "(TimeSheetStatus eq '10' or " +
+        "TimeSheetStatus eq '20' or " +
+        "TimeSheetStatus eq '30' or " +
+        "TimeSheetStatus eq '40' or " +
+        "TimeSheetStatus eq '50')"
+
+    let filter = `${dateFilter} and ${statusFilter}`
+
+    let url = `API_MANAGE_WORKFORCE_TIMESHEET/TimeSheetEntryCollection?$filter=${filter}`
+
+    console.log('## URL -> ' + url)
+
+    let respGET = await client.fetch(url,
+        {
+            method: 'GET',
+            headers: headersGET
+        });
+
+    if (!respGET.ok) {
+        throw new Error(`Could not Retrieve Time Entries (${respGET.status}: ${respGET.statusText})`);
+    }
+
+
+    let timeEntryResponse = await respGET.json()
+
+    for (let i = 0; i < timeEntryResponse.d.results.length; i++) {
```

Comment:
combination of `for` + `Array.push` is sign that it could be replaced by `Array.map`